include/deal.II/dofs/dof_handler.h

```diff
@@ -707,8 +702,9 @@ class DoFHandler  :  public Subscriptor
    * described in the section on the @p make_boundary_sparsity_pattern
    * function.
    */
+  template<typename number>
   types::global_dof_index
-  n_boundary_dofs (const FunctionMap &boundary_ids) const;
+  n_boundary_dofs (const typename dealii::FunctionMap<spacedim,number>::type &boundary_ids) const;
```

Comment:
This doesn't work. `number` is not deducible in the current context, so you force every user to explicitly specify the template argument. This is not backward compatible, nor convenient.

Just remove the `FunctionMap` altogether and use the underlying `::type` typedef. You may want to say in the documentation of the class that the type of `boundary_ids` equals `typename FunctionMap<spacedim,number>::type`.